test/range/count.wren

```diff
@@ -0,0 +1,25 @@
+// Ordered range.
+IO.print((2..5).count) // expect: 4
+IO.print((3..3).count) // expect: 1
+IO.print((0..3).count) // expect: 4
+IO.print((-5..3).count) // expect: 9
+IO.print((-5..-2).count) // expect: 4
+
+// Backwards range.
+IO.print((5..2).count) // expect: 4
+IO.print((3..0).count) // expect: 4
+IO.print((3..-5).count) // expect: 9
+IO.print((-2..-5).count) // expect: 4
+
+// Exclusive ordered range.
+IO.print((2...5).count) // expect: 3
+IO.print((3...3).count) // expect: 0
+IO.print((0...3).count) // expect: 3
+IO.print((-5...3).count) // expect: 8
+IO.print((-5...-2).count) // expect: 3
+
+// Exclusive backwards range.
+IO.print((5...2).count) // expect: 3
+IO.print((3...0).count) // expect: 3
+IO.print((3...-5).count) // expect: 8
+IO.print((-2...-5).count) // expect: 3
```

Comment:
I think to explain you just need to look at the values. `Range` will iterate with a step size of 1, starting from the `from` value and ending at the `to` value (either inclusive or exclusive).

So a range of `(3.1)..(6.1)` goes through the values `3.1, 4.1, 5.1, 6.1`. The last value here is the `to` value, but since the range is "inclusive" this value is still considered part of the range. In contrast `(3.1)...(6.1)` is "exclusive" so once it sees 6.1 it will consider its range exhausted and stop iterating, yielding only the values `3.1, 4.1, 5.1`.

The way to calculate the number of elements in the range could be:
```c++
double delta = std::abs(to - from);
double count = std::floor(delta);
if (exclusive && delta == count)
    count -= 1.0;
```
Since whenever `delta` is not a whole number, it does not matter whether the range is exclusive or inclusive because the last value is never hit anyway. But when the delta is a whole number, it will hit exactly the last value and then the exclusive range is one element shorter.